modules/tinymce/src/themes/silver/main/ts/ui/core/complex/BespokeNumberInput.ts

```diff
@@ -0,0 +1,126 @@
+import { AddEventsBehaviour, AlloyComponent, AlloyEvents, Behaviour, Button, Focusing, Input, Keying, NativeEvents, Representing, SketchSpec } from '@ephox/alloy';
+import { Cell, Fun, Id, Optional } from '@ephox/katamari';
+import { Focus, SugarElement, Traverse } from '@ephox/sugar';
+
+import Editor from 'tinymce/core/api/Editor';
+import { UiFactoryBackstage } from 'tinymce/themes/silver/backstage/Backstage';
+
+import { onControlAttached, onControlDetached } from '../../controls/Controls';
+import { updateMenuText, UpdateMenuTextEvent } from '../../dropdown/CommonDropdown';
+import { onSetupEvent } from '../ControlUtils';
+import { NumberInputSpec } from './FontSizeBespoke';
+
+interface BespokeSelectApi {
+  readonly getComponent: () => AlloyComponent;
+}
+
+const createBespokeNumberInput = (editor: Editor, backstage: UiFactoryBackstage, spec: NumberInputSpec): SketchSpec => {
+  const currentValue = Cell('');
+  const currentComp: Cell<Optional<AlloyComponent>> = Cell(Optional.none());
+
+  const onSetup = onSetupEvent(editor, 'NodeChange', (api: BespokeSelectApi) => {
+    const comp = api.getComponent();
+    currentComp.set(Optional.some(comp));
+    spec.updateText(comp);
```

Comment:
I realise it's what we've called it elsewhere, but using `updateText` for a function that doesn't take the value to update it to keeps causing me problems as I read this. Is there a name that we can use (assuming it's not public-facing) that suggests that the function itself works out the value, rather than being provided the value as an argument?